contracts/ConverterRamp.sol

```diff
@@ -0,0 +1,383 @@
+pragma solidity 0.5.0;
+
+import './interfaces/Cosigner.sol';
+import './interfaces/diaspore/DebtEngine.sol';
+import './interfaces/diaspore/LoanManager.sol';
+import './interfaces/token/Token.sol';
+import './interfaces/token/TokenConverter.sol';
+import './interfaces/RateOracle.sol';
+import './math/SafeMath.sol';
+import './ownership/Ownable.sol';
+
+
+contract ConverterRamp is Ownable {
+    using SafeMath for uint256;
+
+    address public constant ETH_ADDRESS = address(0x00eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee);
+    uint256 public constant AUTO_MARGIN = 1000001;
+    // index of convert rules for pay and lend
+    uint256 public constant I_MARGIN_SPEND = 0;    // Extra sell percent of amount, 100.000 = 100%
+    uint256 public constant I_MAX_SPEND = 1;       // Max spend on perform a sell, 0 = maximum
+    uint256 public constant I_REBUY_THRESHOLD = 2; // Threshold of rebuy change, 0 if want to rebuy always
+    // index of loan parameters for pay and lend
+    uint256 public constant I_LOAN_MANAGER = 0;     // Loan Manager contract
+    uint256 public constant I_REQUEST_ID = 1;      // Loan id on Diaspore
+    // for pay
+    uint256 public constant I_PAY_AMOUNT = 2; // Amount to pay of the loan
+    uint256 public constant I_PAY_FROM = 3;   // The identity of the payer of loan
+    // for lend
+    uint256 public constant I_LEND_COSIGNER = 2; // Cosigner contract
+    // address diaspore model
+    uint256 public constant I_DEBT_ENGINE = 4; //
+
+    event RequiredRebuy(address token, uint256 amount);
+    event Return(address token, address to, uint256 amount);
+    event OptimalSell(address token, uint256 amount);
+    event RequiredRcn(uint256 required);
+    event RunAutoMargin(uint256 loops, uint256 increment);
+
+    function() external payable {
+        require(msg.value > 0, 'The value is 0.');
+    }
+
+    function pay(
```

Comment:
reveer si es la mejor solucion con diaspore

Reply:
You can replace `executeOptimalPay` by the native method of Diaspore `payTokens`